### DISCUSSION TITLE
RTK Query streaming updates - chamging the order of operations

### DISCUSSION THREAD

**Discussion by mdeperas** [reduxjs/redux-toolkit#3379] [Q&A]
Hi, I've read the [docs](https://redux-toolkit.js.org/rtk-query/usage/streaming-updates#using-the-oncacheentryadded-lifecycle) for streaming updates and I understand the advised approach. 
As Mark Erikson responsed in my previous discussion thread, typically, you will await cacheDataLoaded to determine when the first data has been fetched, then use the updateCacheData utility to apply streaming updates as messages are received.

But I am trying to find out if it is possible to apply opposite order of operations.

1) First start listening to streaming updates,
2) Then make the http request and feed the cache.
Is there any advised approach for such operation?

Thanks for any help.

**Comment by phryneas** [comment]
You could try to use a `queryFn`, create the subscription in there and somehow pass it to `onQueryStarted` through the `meta` field, but as far as I know you'd be the first person to try that.

**Reply by mdeperas** [reply]
Thanks @phryneas for the answer. So the gap between the initial request/response and the moment I will start listening to the streaming updates should not be considered as a problem? 

I am wondering what if a message from streaming update will be unhandled because, response to the request did not returned and I am not yet listening to the streaming updates.
